test/common/config/registry_test.cc

```diff
@@ -103,6 +103,17 @@ TEST(RegistryTest, DEPRECATED_FEATURE_TEST(WithDeprecatedFactoryPublished)) {
                           ->name());
 }
 
+class TestWithDeprecatedPublishedFactoryNoDeprecatedNames : public PublishedFactory {
+public:
+  std::string name() const override { return ""; }
+};
+
+TEST(RegistryTest, DEPRECATED_FEATURE_TEST(AssertsIfNoDeprecatedNamePassed)) {
```

Comment:
Does this test case depend on any deprecated feature?

Also, it's confusing that the test name mentions "Passed" but the test body includes EXPECT_DEATH.  Can you add a comment describing what this test does?

Reply:
The tests are based slightly off of VersionedWithDeprecatedNamesFactory and WithDeprecatedFactoryPublished which both do use DEPRECATED_FEATURE_TEST. I might be missing additional context to know if the usage is justified.

Reply:
Briefly looking at https://github.com/envoyproxy/envoy/pull/10103/commits (where tests within this file started to get annotated with DEPRECATED_FEATURE_TEST), I think the usage makes sense as it's a test using deprecated names and it's possible via a flag to turn off extensions using deprecated names.

See: https://github.com/envoyproxy/envoy/pull/10103/commits/0fa82827bd3251b78ed4fbb2b8d6607b43bfc3d4#diff-ff1cd959484d987679938efe34a0dd93 for additional context.